### ISSUE TITLE
Big Compose Field

### ISSUE THREAD

**Issue by remino** [mastodon/mastodon#26603]
### Pitch

Hi,

In both the normal and advanced Web desktop interfaces, I find the compose field incredibly small. I think it could be enlarged when needed.

I wrote a userscript for myself which “pops out” the compose form once it gets focus:
https://github.com/remino/userscripts/raw/main/userscripts/mastodon-big-compose.user.js

After installing it, the compose form is enlarged and moved to the centre of the screen while composing:

<img width="410" alt="image" src="https://github.com/mastodon/mastodon/assets/29999/9924a8b9-e075-4b25-90d6-be90334a5a00">

I use it, but it’s a bit buggy. It’s more like a proof of concept than something usable by everyone.

Also, I’m certain not everyone wants this. Maybe it’s something users could choose to enable.

### Motivation

Posts can get quite long. Having a large input field would make it easier to see what you’re typing, especially when it gets long or when users add media. Other social platforms give you a big nice in-your-face field when you write a post, and I think Mastodon users deserve that too.

**Comment by unseenlarks** [comment]
it could be dragged down/enlarged as needed, much like GitHub's comments field allows.

**Comment by mato** [comment]
Yes, please! Composing in a box sized roughly 6x3cm on my screen is a pain. Having recently jumped onto Mastodon from Twit^H^H^H^HX I went looking for a setting for this and was disappointed not to find one.

**Comment by simon-frankau** [comment]
The input field can get pretty tiny (more-or-less one line high!) when replying to a long post, which can be very frustrating considering the amount of screen real estate that might otherwise be available.

I would find a "pop-out" input form very helpful, but do suggest to make it easy to see the post (or ideally thread) you're replying to at the same time.

**Comment by brendanjones** [comment]
Definitely agree on the problem, the compose box is annoyingly small. 

One downside of a modal for composing posts, however, is that you can't touch any other text on the page while you're composing your post. So if you're referencing another post then you can't copy+paste into yours. So yes, it's nice that it's big but there is a downside. Making it a user preference is definitely a good idea.

Even if the modal isn't added, If the compose box padding is removed and the column made 300px then the compose box becomes a little wider. That's at least an improvement: https://github.com/mastodon/mastodon/issues/19816

**Comment by telephon** [comment]
It would also help if the columns were adustable. I think ist is possible with plain HTML: https://www.brainbell.com/javascript/making-resizable-table-js.html
